modules/table/src/integrationTest/java/org/apache/ignite/distributed/ItTxDistributedTestSingleNode.java

```diff
@@ -148,6 +148,7 @@ public void before() throws Exception {
     @AfterEach
     public void after() throws Exception {
         txTestCluster.shutdownCluster();
+        Mockito.framework().clearInlineMocks();
```

Comment:
1. I have no such examples, but you can dig tests which fails with OOM, for example ItTxDistributedTestSingleNode
2. Yes, it's possible, but be aware of leaks per suite.
3. Because it is per method, not per class. Moving the call to clearInlineMocks at @aftereach is exactly the thing to do, but should be done as separate ticket. I've created the one [1]
[1] https://issues.apache.org/jira/browse/IGNITE-21049

> In any case, it would be nice to split such Mockito related stuff from lock enhancements 
it is not possible. Without this stuff TC run ends with OOM.